lib/logflare/backends/adaptor/slack_adaptor.ex

```diff
@@ -39,17 +43,50 @@ defmodule Logflare.Backends.Adaptor.SlackAdaptor do
 
   ### Example
 
-    iex> %{blocks: [block]} = to_body([%{"some" => "key"}])
-    iex> get_in(block, [:text, :text])
-    "• some: key"
+    iex> %{blocks: [_]} = to_body([%{"some" => "key"}])
+    iex> %{blocks: [_, _]} = to_body([%{"some" => "key"}], context: "some context")
 
   """
   @spec to_body([map()]) :: map()
-  def to_body(results) when is_list(results) do
+  def to_body(results, opts \\ []) when is_list(results) do
+    context = Keyword.get(opts, :context)
+    button_link = Keyword.get(opts, :button_link)
+
     %{
-      blocks: [
-        %{type: "section", text: %{type: "mrkdwn", text: to_markdown(results)}}
-      ]
+      blocks:
+        ([
+           if(context != nil,
+             do: %{
+               type: "context",
+               elements: [%{type: "mrkdwn", text: context}]
+             }
+           )
+         ] ++
+           Enum.map(results, fn row ->
+             %{
+               type: "rich_text",
+               elements: [
+                 %{
+                   type: "rich_text_preformatted",
+                   elements: to_rich_text_preformatted(row)
+                 }
+               ]
+             }
+           end) ++
+           [
+             if(button_link != nil,
+               do: %{
+                 type: "section",
+                 accessory: %{
+                   type: "button",
+                   text: %{type: "plain_text", text: button_link.text},
+                   url: button_link.url,
+                   style: "primary"
+                 }
+               }
+             )
+           ])
+        |> Enum.filter(&(&1 != nil))
```

Comment:
Ok, that looks awful. Maybe some extraction to some helper functions would make it easier to read.